js/compiler/compiler.go

```diff
@@ -48,42 +48,58 @@ var (
 	once             sync.Once
 )
 
+// CompatibilityMode specifies the JS compatibility mode
+// nolint:lll
+//go:generate enumer -type=CompatibilityMode -transform=snake -trimprefix CompatibilityMode -output compatibility_mode_gen.go
+type CompatibilityMode uint
+
+// "es6": default, Goja+Babel+core.js
+// "es51": plain Goja
+const (
+	CompatibilityModeES6 CompatibilityMode = iota
+	CompatibilityModeES51
+)
+
 // A Compiler uses Babel to compile ES6 code into something ES5-compatible.
 type Compiler struct {
 	vm *goja.Runtime
 
 	// JS pointers.
-	this      goja.Value
-	transform goja.Callable
-	mutex     sync.Mutex //TODO: cache goja.CompileAST() in an init() function?
+	this              goja.Value
+	transform         goja.Callable
+	mutex             sync.Mutex //TODO: cache goja.CompileAST() in an init() function?
+	compatibilityMode CompatibilityMode
 }
 
-// Constructs a new compiler.
-func New() (*Compiler, error) {
+// New constructs a new compiler with the provided compatibility mode
+func New(compatMode CompatibilityMode) (*Compiler, error) {
```

Comment:
Should we use functional Option for this? The code above said default is Goja+Babel+core.js, but here we must always pass a mode to New.

Reply:
What do you mean with "functional Option"?

Maybe the `CompatibilityMode` comment is out of place here, since it's meant more for a user of `NewBundle*` than `Compiler`. I'll move it to somewhere in `js` or `lib` wherever we do this validation.

Reply:
@imiric https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
Ah, variadic args. Nah, I would rather not do that in this case.

`compatibilityMode` should be a required Compiler argument: we can't run without it, and it should be a conscious decision by the user (`NewBundle*` in this case) in which mode to run in. The only reason we assume ES6 as default is to preserve backwards compatibility with the CLI.

I disagree with you here, but we can discuss it further if you feel strongly about it.

Reply:
@imiric then you should not mention "default is ES6 ..." in above comment. Why default since when you force user to give an input?

Reply:
Right, that's why I said I would move the comment. :) Would that resolve this?